resources/schema.py

```diff
@@ -492,7 +464,21 @@ class Input(BerthInput):
     @add_permission_required(Berth)
     @transaction.atomic
     def mutate_and_get_payload(cls, root, info, **input):
-        input["berth_type"] = get_berth_type(info, input)
+        width = input.pop("width", None)
+        length = input.pop("length", None)
+        depth = input.pop("depth", None)
+        mooring_type = input.pop("mooring_type", None)
+
+        if all([width, length, mooring_type]):
```

Comment:
I guess this `if` won't be necessary, since dimensions and mooring type should be required (except for `depth` of course)